### ISSUE TITLE
TypeError: 'set' on proxy: trap returned falsish for property 

### ISSUE THREAD

**Issue by jayesbe** [ParmesanPangolin/electron-proxy-store#1]
Hi, thanks for this package.. it looks like it would be super useful but it doesn't seem usable at the moment.

I continuously get the error

TypeError: 'set' on proxy: trap returned falsish for property 

it simply fails to save on this line

`
store.data.server = this.server;
`

What's funny is that it appears to save the value because if I reload electron it reloads the value if I call it. Storing a value immediately crashes

Am I doing something wrong?

**Comment by mendellerner** [comment]
Thanks for taking an interest.

---
**tldr;** functions (`array.push()`) or instances (`http.createServer()`) work differently under the hood from regular `=`. Therefore returning a sort of falsy value to the proxy even though everything still works correctly.

---

This is a problem that I had as well when creating the proxy (which sets the variables).

Firstly, if you could send me a larger segment of your code (from when you first set and initialize the server), I may be able to give you a better answer, and possibly solve the TypeError completely. Read on for why.

In lieu of more of your code, I'm going to assume that server is an http server of the form `http.createServer()`. This is important because it is not a static variable, rather it's an instance.

I did not have a lot of time to look into this issue, but what I found was that there are 2 parts to the proxy, 1) what the proxy does, and 2) an implicit "return value" into the proxy itself.

Even when the set proxy throws a type error, the program still continues as if the error didn't occur, meaning that it sets the variable as you would expect.

The code that I had been experimenting with that produced this error was (for example):

```js
store.data.myArray = []
store.data.myArray[1] = 'no error'
// => Produces no error

store.data.myArray.push('error')
// => TypeError: 'set' on proxy: trap returned falsish for property

console.log(store.data.myArray)
// => ['no error', 'error']
```

It appears, from my limited research, that setting variables using the regular `=` always returns a value of `true` under the hood (that the set worked correctly), which the proxy expects. However, when using a **function** to set the variable, such as `array.push()` or in your case `store.data.server = this.server` which again, is not static variable, two things happen:
1) the function input into the proxy runs it's own set function which sets correctly to the `store` and returns the truthy value expected by the proxy under the hood.
2) The function originally sent to the store (I.e. `array.push` or `this.server`) are functions and therefore are not set the way a regular value is. They therefore return a falsy value, which produces the error you are seeing.

If you send me your code, I may be able to investigate this further, time permitting.
